src/reference/workers/taskcluster-worker/engine-plugin.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Engine-Plugin interaction
+order: 40
+sequence_diagrams: true
+---
+
+Putting this all together, we get the following interactions.
```

Comment:
Yeah, now that I look at it, this really isn't documentation so much as some design notes (I don't even know if they're still current..), so i can move it to a single page under that title.